scipy/stats/_continuous_distns.py

```diff
@@ -3582,56 +3583,943 @@ class levy_stable_gen(rv_continuous):
+    ..math::
+        \beta_B K(\alpha) = \frac{2}{\pi} atan(\beta_A \tan[\pi\alpha / 2])
+        c_B = \frac{c_A}{\cos[\beta_B K(\alpha)\pi/2]}
+        mu_B = mu_A \cos[\beta_B K(\alpha)\pi/2]
+
+    - Parameterization C:
+    .. math::         
+        \varphi(t; \alpha, \delta, c_C, \mu_C) = e^{ c_C(it\mu_C -|t|^{\alpha}e^{-i\alpha \delta \operatorname{sign}(t)\pi/2}}
+
+    where:
+              
+    ..math::
+         \delta =\frac{\beta_B K(alpha)}{\alpha} = \frac{2}{\alpha \pi} atan(\beta_A \tan[\pi\alpha / 2])
+         c_C = c_B = \frac{c_A}{\cos[\beta_B K(\alpha)\pi /2}
+ 
+    - Parameterization P:
+    .. math::         
+        \varphi(t; \alpha, p_1, c_P, \mu_P) = e^{ c_P(it\mu_P -|t|^{\alpha}e^{ -i\alpha (p_1 - p_2) \operatorname{sign}(t)\pi/2}}
+
+    where:
+              
+    ..math::
+        p_1 + p_2 = 1 and 
+        0 <= p_1, p_2 <= 1 for 0 < \alpha < 1, 
+        1 / \alpha <= p_1, p_2 <= 1 - 1 / \alpha for 1 < \alpha < 2. 
+
+        p_1 = \frac{1 + \delta}{2} = \frac{1}{2} + \frac{\beta_B K(alpha)}{2\alpha} = 
+            \frac{1}{2} + \frac{1}{\alpha \pi} atan(\beta_A \tan[\pi\alpha / 2])
+         c_P = c_C = c_B = \frac{c_A}{\cos[\beta_B K(\alpha)\pi /2}
+
+    The parameterization P has rather intuitive relation to CDF of stable law. 
+    The parameter p_1 = 1 - CDF(0, \alpha, p_1, c), i.e. the parameter p_1 gives value 
+    of the size of the probability mass on the positive real line and therefore gives 
+    valuable information about the asymmetry of the distribution. 
+
+
+
+    
+    References
+    ----------    
+    .. [ZOL1] Zolotarev V.M., 1993. On representation of densities of stable laws by special functions. 
+       SIAM ... 
+    .. [ZOL2] Zolotarev V.M., 1986. One-dimensional stable distributions.
+       Amer. Math. Soc., Providence, RI.
+    .. [BE] Bergstrom H., 1952. On some expansions of stable distribution functions. 
+       Arkiv for Matematik 2, 4, p.375-378
+       [NO] Nolan J.P., 1997. Numerical calculations of stable densities and distribution functions. 
+       Commun Statist. - Stochastic Models, 13(4), p.759-774
+    .. [MC] McCulloch, J., 1986. Simple consistent estimators of stable distribution parameters.
+       Commun. Statist.–Simula. 15(4) 1109–1136
+       [KO] Koutrouvelis, I. A., 1980. Regression–Type Estimation of the Parameters of Stable Laws.
+       J. Amer. Statist. Assoc. 75, p.918–928.
+    .. [MS] Mittnik, S.T. Rachev, T. Doganoglu, D. Chenyao, 1999. Maximum likelihood estimation 
+       of stable Paretian models, Mathematical and Computer Modelling, Volume 29, Issue 10, 
+       1999, Pages 275-293.
+    .. [WE] Weron A, Weron R., 1995. Computer Simulation of Levy alpha-stable variables and processes. 
+       Chaos - The interplay between stochastic and deterministic behaviour, Springer, p.545 - 558  
+
+
+    %(example)s
 
     """
+    supported_parameterizations = frozenset(["A", "B", "C", "P"])
 
-    def _rvs(self, alpha, beta):
+    @classmethod
```

Comment:
_argcheck should be an instance method, since distribution objects are instances